compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -793,6 +836,14 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
     }
     countRefs.traverse(tree)
     for (binding <- bindings) countRefs.traverse(binding)
+
+    def retain(boundSym: Symbol) = {
+      refCount.get(boundSym) match {
+        case Some(x) => x > 1 || x == 1 && !boundSym.is(Method)
+        case none => true
+      }
+    } && !boundSym.is(TransparentImplicitMethod)
```

Comment:
I doubt the "more performant part" since the test is`boundSym.is(TransparentImplicitMethod)` is ~99% likely to succeed. So it's better to put the test first that prunes more cases.